### DISCUSSION TITLE
Resolver input type validation is to strict

### DISCUSSION THREAD

**Discussion by karlismelderis** [react-hook-form/react-hook-form#5483] [Q&A]
**Describe the bug**
🙏 I'm sorry if I'm missing some hidden config.
We have two YUP inferred types - data from API (with loads of info missing) and one for validation.
Starting type is much more loose and permit nulls.

but yupResolver doesn't like it 
```ts
  const SourceSchema = object({
    valueA: boolean(),
  });
  const SubmitSchema = object({
    valueA: boolean().required(),
  });
  const valuesTest: InferType<typeof SourceSchema> = {valueA: true};

  yupResolver(SubmitSchema)({...valuesTest});
```

gives error:
```
TS2345: Argument of type '{ valueA: boolean | undefined; }' is not assignable to parameter of type '{ valueA: boolean; }'. 
  Types of property 'valueA' are incompatible.
     Type 'boolean | undefined' is not assignable to type 'boolean'.
       Type 'undefined' is not assignable to type 'boolean'.
```
similar issue happens if null is possible starting point.

Changing Resolver type a bit solved the case.
Changed values -> `values: Record<string, any>`

```ts
export declare type Resolver<TFieldValues extends FieldValues = FieldValues, TContext extends object = object> = (values: Record<string, any>, context?: TContext, validateAllFieldCriteria?: boolean) => Promise<ResolverResult<TFieldValues>> | ResolverResult<TFieldValues>;
```

As I see it it should be fine to put "any" data as values and `yupResolver` should tell if data matches provided schema.

**Comment by bluebill1049** [comment]
can you provide a codesandbox?
